### PR TITLE
dotCMS/core#17722 Show submenu as flyout when menu is collapsed 

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/view/components/dot-navigation/components/dot-nav-item/dot-nav-item.component.spec.ts [hunk 1]
```diff
@@ -122,51 +107,21 @@ describe('DotNavItemComponent', () => {
...
-                expect(iframeOverlayService.hide).toHaveBeenCalledTimes(1);
-            });
-            it('should close on document click', () => {
-                document.dispatchEvent(new MouseEvent('click'));
-                expect(component.contextmenu).toBe(false);
-            });
+        it('should emit menuRightClick on right click', () => {
+            spyOn(component.menuRightClick, 'emit');
+            de.triggerEventHandler('contextmenu', new MouseEvent('contextmenu'));
+            expect(component.menuRightClick.emit).toHaveBeenCalledTimes(1);
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/components/dot-nav-item/dot-nav-item.component.spec.ts]: You need to check what is this event contains.

File: src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <ul class="dot-nav-sub"
-    [@expandAnimation]="data.isOpen ? 'expanded' : 'collapsed'"
-    [class.contextmenu]="contextmenu"
+    [@expandAnimation]="( data.isOpen && !collapsed ) ? 'expanded' : 'collapsed'"
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html]: Do we unit test this?

File: src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+    [class.contextmenu]="data.isOpen && collapsed"
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html]: Do we unit test this?

File: src/app/view/components/dot-navigation/services/dot-navigation.service.spec.ts [hunk 4]
```diff
@@ -209,9 +209,17 @@ describe('DotNavigationService', () => {
...
                 counter++;
             });
+            service.collapseMenu();
+        });
+    });
 
+    describe('collapseMenu', () => {
+        it('should collapse menu and call closeAllSections', () => {
+            expect(service.collapsed$.getValue()).toBe(true);
+            spyOn(service, 'closeAllSections');
```
Comment (fmontes) [inline src/app/view/components/dot-navigation/services/dot-navigation.service.spec.ts]: We can't spy on a method of the service we're testing, we need to test what happen when that method gets called.

File: src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html (hunk 2 above)
Comment (hmoreras) [reply src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html]: done

File: src/app/view/components/dot-navigation/services/dot-navigation.service.spec.ts (hunk 4 above)
Comment (hmoreras) [reply src/app/view/components/dot-navigation/services/dot-navigation.service.spec.ts]: done

File: src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html (hunk 3 above)
Comment (hmoreras) [reply src/app/view/components/dot-navigation/components/dot-sub-nav/dot-sub-nav.component.html]: done

**Pull request merged**